lib/minheap.js

```diff
@@ -38,26 +56,72 @@ class MinHeap {
   }
 
   // This method returns true if the heap is empty
-  // Time complexity: ?
-  // Space complexity: ?
+  // Time complexity: O(1)
+  // Space complexity: O(1)
   isEmpty() {
-    throw new Error("Method not implemented yet...");
+    if (this.store.length === 0) return true
+
+    return false;
   }
 
   // This helper method takes an index and
   //  moves it up the heap, if it is less than it's parent node.
   //  It could be **very** helpful for the add method.
-  // Time complexity: ?
-  // Space complexity: ?
+  // Time complexity: O(log n)
+  // Space complexity: O(1)
   heapUp(index) {
-    throw new Error("Method not implemented yet...");
+    // update position in heap/array
+    const parentIndex = Math.floor((index - 1) / 2);
+
+    const element = this.store[index];
+    const parent = this.store[parentIndex];
+
+    // base case
+    if (index === 0 || parent.key < element.key) {
+      // pass
```

Comment:
This works however, it's generally considered bad style to leave the body of an if statement empty
```suggestion
      return
```